### PR TITLE
Alerting: Use Adaptive Cards in Teams notifications

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What this PR does / why we need it**:

This pull request changes the cards in Teams notifications from Office 365 Connector cards to Adaptive Cards to fix an issue where images were not shown in Teams for desktop and web. Since Office 365 Connector cards are deprecated, it made sense to move to Adapative Cards and fix this bug at the same time.

The Adaptive Card messages maintain the design of the Office 365 Connector Card with a number of minor differences:

- In Adaptive Card messages the color of the title is red or green depending on the status of the alerts, where as with Office 365 connector cards there was a colored border at the top of the title

- In Adaptive Card messages the title is bold to make it easier to read when the color is red or green

- In Adaptive Card messages the thumbnails for images are medium size if there are more than two images, otherwise large size

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

# Release notice breaking change

Alert notifications to Microsoft Teams now use Adaptive Cards instead of Office 365 Connector Cards.

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/teams.go [hunk 1]
```diff
@@ -1,31 +1,208 @@
...
+}
+
+// AdaptiveCardsAttachment contains an adaptive card.
+type AdaptiveCardsAttachment struct {
+	Content     AdaptiveCard `json:"content"`
+	ContentType string       `json:"contentType"`
+	ContentURL  string       `json:"contentUrl,omitempty"`
+}
+
+type AdaptiveCard struct {
```
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/channels/teams.go:68]:
```suggestion
// AdaptiveCard is the top level schema for an adaptive card. See: https://adaptivecards.io/explorer/AdaptiveCard.html
type AdaptiveCard struct {
```

Found a nice schema description, might help in the future.

**Review by JacobsonMT — APPROVED** [review]
Great work! One nit suggestion to add a schema url, otherwise LGTM :rocket: 

We might want to mention the deprecation of support for Office 365 Connectors somewhere though.

Here's a screenshot for future readers:

image

**Comment by tfreycb** [comment]
We updated to 9.1 today and noticed that the Teams notifications on the mobile phone and in the MS Teams activity history now say "Card" instead of the notification title. We have custom notification templates, is there anything that needs to change in the template to revert this behavior as it makes the phone notifications rather useless.

**Comment by grobinson-grafana** [comment]
Hi! 👋 I have been able to reproduce this issue and have opened a fix that should be available in Grafana 9.1.1. https://github.com/grafana/grafana/pull/53810

**Comment by MajedFarhat** [comment]
can I use it to add mention in the message

**Pull request merged**